code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -730,25 +742,100 @@ func (fsh *StorageHandler) GetRefs(ctx context.Context, r *http.Request) (*blobb
 		return nil, common.NewError("invalid_parameters", "Invalid allocation id passed."+err.Error())
 	}
 
+	clientID := ctx.Value(constants.ContextKeyClient).(string)
+	if len(clientID) == 0 {
+		return nil, common.NewError("invalid_operation", "Client id is required")
+	}
+
+	publicKey, _ := ctx.Value(constants.ContextKeyClientKey).(string)
```

Comment:
is it better to says it is required like clientId? why client doesn't send it?

Reply:
@cnlangzi Any suggestion?
Client may forget to send `client id`. Today we use gosdk which sends it by default. There may be other application that don't use gosdk( and have their own sdk).

Reply:
if he forget to send it, throw error may be better.

Reply:
do you mean we should instead return "invalid client id"?